src/js/entities-service/entities/actions.js

```diff
@@ -250,6 +250,14 @@ const _Model = BaseModel.extend({
 
     return !!size(programAction.get('allowed_uploads'));
   },
+  removeAttachment(resource) {
+    const files = this.get('_files');
+    const newFilesRelationship = files.filter(file => {
+      return file.id !== resource.id;
+    });
+
+    this.set({ _files: newFilesRelationship });
+  },
```

Comment:
The paperclip attachments icon on action list items is shown or hidden based on the length of the `action.get('_files')`.

We're currently not updating that when a file is removed manually by a user. And therefore it doesn't work out of the box for the `FileRemoved` ws notification.

So i added this method to action entity service that filters out the file from `_files` by `id`. And then resets the `_files` relationship data for the action.

This allows for the paperclip icon to be hidden after attachment deletion.

Reply:
If this is how we want to implement this, we would just need to call `action.removeAttachment(model)` in the action sidebar app when a file is manually deleted by a user.

Reply:
Currently in the app, the `.paperclip` icon isn't be shown when an attachment is added and isn't hidden when an attachment is removed manually by a user in the sidebar.

I created a separate story [here]( https://app.shortcut.com/roundingwell/story/58034/in-action-list-items-show-hide-paperclip-icon-when-an-attachment-is-manually-added-removed-via-action-sidebar), that I can tackle in a separate PR.